pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -57,26 +57,69 @@ var snapshotIngestAsWriteThreshold = settings.RegisterByteSizeSetting(
 	}())
 
 // replicaRaftStorage implements the raft.Storage interface.
+//
+// All mutating calls to raft.RawNode require that r.mu is held. All read-only
+// calls to raft.RawNode require that r.mu is held at least for reads.
+//
+// All methods implementing raft.Storage are called from within, or on behalf of
+// a RawNode. When called from within RawNode, r.mu is held necessarily (and
```

Comment:
Minor, but would be good to understand if reads can be made with `r.mu.RLock` only. Maybe not, because of the implementation details like https://github.com/cockroachdb/cockroach/blob/5cba64b451d6d12eac95aee65e49f6e2c4dc5084/pkg/kv/kvserver/logstore/stateloader.go#L30-L40.